crates/federate/src/inboxes.rs

```diff
@@ -38,15 +38,12 @@ static FOLLOW_REMOVALS_RECHECK_DELAY: LazyLock<chrono::TimeDelta> =
 
 #[async_trait]
 pub trait DataSource: Send + Sync {
-  async fn read_site_from_instance_id(
-    &self,
-    instance_id: InstanceId,
-  ) -> Result<Option<Site>, diesel::result::Error>;
+  async fn read_site_from_instance_id(&self, instance_id: InstanceId) -> LemmyResult<Option<Site>>;
```

Comment:
This one too could just go to `LemmyResult<Site>`, and if you need to call `.ok()`, you can do it where this is called.

Reply:
Changed everything you mentioned.